src/components/UpgradeEnvironmentModal/UpgradeEnvironmentModal.tsx

```diff
@@ -0,0 +1,44 @@
+import { BuildConfiguration } from 'pnc-api-types-ts';
+
+import { useServiceContainer } from 'hooks/useServiceContainer';
+
+import { ActionModal } from 'components/ActionModal/ActionModal';
+
+import * as buildConfigApi from 'services/buildConfigApi';
+
+export interface IUpgradeEnvironmentModalProps {
+  isModalOpen: boolean;
+  toggleModal: () => void;
+  buildConfig: BuildConfiguration;
+}
+
+export const UpgradeEnvironmentModal = ({ isModalOpen, toggleModal, buildConfig }: IUpgradeEnvironmentModalProps) => {
+  const serviceContainerUpgradeEnvironment = useServiceContainer(buildConfigApi.upgradeEnvironment, 0);
+
+  const confirmModal = () => {
+    serviceContainerUpgradeEnvironment
+      .run({
+        serviceData: {
+          buildConfigId: buildConfig.id,
+          replacementEnvironmentId: buildConfig.environment!.attributes!.DEPRECATION_REPLACEMENT,
+        },
+      })
+      .catch(() => {
+        console.error('Failed to close Product Milestone.');
+      });
+  };
+
+  return (
+    <ActionModal
+      modalTitle={`Upgrade environment ${buildConfig.environment?.name}?`}
+      actionTitle="Upgrade"
+      isOpen={isModalOpen}
+      onToggle={toggleModal}
+      onSubmit={confirmModal}
+      serviceContainer={serviceContainerUpgradeEnvironment}
+      modalVariant="medium"
+    >
+      Environment will be upgraded to the latest replacement.
```

Comment:
Include both environments and display full Environment details so that the user can verify he is fine with the new environment before upgrading: 
1) from (old)
2) to (new)

See jira comment for more details.